### DISCUSSION TITLE
Disable file system access API usage by default

### DISCUSSION THREAD

**Discussion by rolandjitsu** [react-dropzone/react-dropzone#1186] [Polls]
Hi everyone,

I just wanted to start a discussion about a change (https://github.com/react-dropzone/react-dropzone/pull/1123 and https://github.com/react-dropzone/react-dropzone/pull/1135) that was introduced a while back in order to provide better support for opening a file picker programmatically and a more stable cancel cb.

The support for that API is very limited and still a working draft. On top of that, [Mozilla has marked the API as harmful](https://mozilla.github.io/standards-positions/#native-file-system). Plus, the user is expected to handle the permissions request before using the dropzone when that API is enabled.

So I'm thinking to make the API usage an opt it sort of thing and use the old `trigger a click on the input` way by default until that API is accepted by all major browser and past the draft status.

But I wanna hear what everyone else thinks about it before I make any other breaking changes.

**Comment by VladimirMikulic** [comment]
I appreciate you sharing this with the community @rolandjitsu.
I personally always like to upgrade and use the latest, but in this state, the API doesn't seem to be there yet. (for the majority)

With that being said, I'd would be fine with either decision, but just that all pros and cons/drawback are clearly documented given the importance of this.

**Comment by jhanschoo** [comment]
+1 to disable by default, because I discovered a bug with it in default builds in linux environment (tested in chrome, chromium, in GNOME 42, openbox, KDE; not sure if it matters): paste `window.showOpenFilePicker({ types: [{ accept: { 'image/jpeg': ['.jpg', '.jpeg'] } }] })` into the dev console and no file picker appears. A description must be supplied, and afaict `react-dropzone` does not use that feature (providing an optional description). I've provided more details on it in https://bugs.chromium.org/p/chromium/issues/detail?id=1350487

**Comment by rolandjitsu** [comment]
I know it's been a long time since this poll was posted, but I think it's still relevant. And the consensus is that the feature should be disabled by default.

I will try to make that change within the next few days.

**Reply by rolandjitsu** [reply]
Should land once https://github.com/react-dropzone/react-dropzone/pull/1370 is merged.

**Reply by rolandjitsu** [reply]
https://github.com/react-dropzone/react-dropzone/pull/1370 is merged. The change should be released with v14.2.9.
